src/repositories/ElasticRepository.ts

```diff
@@ -9,4 +10,11 @@ export default class ElasticRepository {
     constructor(client: ElasticWrapper) {
         this.client = client;
     }
+
+    /**
+     * @returns string
+     */
+    protected getType(): string {
```

Comment:
You can use `abstract` classes for this purposes.
Check it here http://www.typescriptlang.org/docs/handbook/classes.html